app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsRepository.kt

```diff
@@ -0,0 +1,100 @@
+package com.keylesspalace.tusky.components.conversation
+
+import androidx.annotation.MainThread
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.paging.Config
+import androidx.paging.toLiveData
+import com.google.gson.Gson
+import com.keylesspalace.tusky.db.AppDatabase
+import com.keylesspalace.tusky.entity.Conversation
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.util.Listing
+import com.keylesspalace.tusky.util.NetworkState
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.util.concurrent.Executors
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ConversationsRepository @Inject constructor(val mastodonApi: MastodonApi, val db: AppDatabase, val gson: Gson) {
+
+    private val ioExecutor = Executors.newSingleThreadExecutor()
```

Comment:
this should probably not be created here

Reply:
Why not, seems file? Can be injected with qualification for testing tho.